instrumentation/failsafe-3.0/testing/src/main/java/io/opentelemetry/instrumentation/failsafe/v3_0/AbstractFailsafeTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.failsafe.v3_0;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import dev.failsafe.CircuitBreaker;
+import dev.failsafe.CircuitBreakerOpenException;
+import dev.failsafe.Failsafe;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.testing.assertj.MetricAssert;
+import java.time.Duration;
+import java.util.Objects;
+import org.junit.jupiter.api.Test;
+
+public abstract class AbstractFailsafeTest {
```

Comment:
A separate testing module is often used to share code between library and javaagent instrumentation. Here you only have the library instrumentation, consider getting rid of the testing module. Also merging the testing code into the library module would probably resolve the latest dep test failure you have.